src/timessquare/storage/nbhtmlcache.py

```diff
@@ -0,0 +1,139 @@
+        parameters : dict
+            The parameter values, keyed by the parameter names, with values as
+            cast Python types
+            (`timessquare.domain.page.PageParameterSchema.cast_value`).
+
+        Returns
+        -------
+        nbhtml : `timessquare.domain.nbhtml.NbHtmlModel`
+            The HTML page domain model.
+        """
+        key = NbHtmlCacheStore.calculate_key(
+            page_name=page_name, parameters=parameters
+        )
+        serialized_data = await self._redis.get(key)
+        if not serialized_data:
+            return None
+
+        return NbHtmlModel.parse_raw(serialized_data.decode())
+
+    async def iter_keys_for_page(self, page_name: str) -> AsyncIterable[str]:
+        """Iterate over all keys for a page.
+
+        Parameters
+        ----------
+        page_name : str
+            The name of the page (corresponds to
+            `timessquare.domain.page.PageModel.name`).
+
+        Yields
+        ------
+        key : str
+            Yields keys for a page.
+        """
+        pattern = r"^{page_name}\/"
+        async for key in self._redis.scan_iter(pattern):
+            yield key
+
+    async def delete(
+        self, *, page_name: str, parameters: Mapping[str, Any]
+    ) -> bool:
+        """Delete the NbHtmlModel of a page for a given set of parameters,
+        if available.
+
+        Parameters
+        ----------
+        page_name : str
+            The name of the page (corresponds to
+            `timessquare.domain.page.PageModel.name`).
+        parameters : dict
+            The parameter values, keyed by the parameter names, with values as
+            cast Python types
+            (`timessquare.domain.page.PageParameterSchema.cast_value`).
+
+        Return
+        ------
+        deleted : bool
+            `True` if a record was deleted, `False` otherwise.
+        """
+        key = NbHtmlCacheStore.calculate_key(
+            page_name=page_name, parameters=parameters
+        )
+        count = await self._redis.delete(key)
+        return count > 0
+
+    @staticmethod
+    def calculate_key(*, page_name: str, parameters: Mapping[str, Any]) -> str:
+        """Create the redis key for an NbHtmlModel given the page's name and
+        parameter values.
+
+        Parameters
+        ----------
+        page_name : str
+            The name of the page (corresponds to
+            `timessquare.domain.page.PageModel.name`).
+        parameters : dict
+            The parameter values, keyed by the parameter names, with values as
+            cast Python types
+            (`timessquare.domain.page.PageParameterSchema.cast_value`).
+
+        Returns
+        -------
+        key : str
+            The unique redis key for this combination of page name and
+            parameter values.
+        """
+        return calculate_key(
+            prefix="nbhtml", page_name=page_name, parameters=parameters
+        )
```

Comment:
This is a minor thing, but having a staticmethod call a function with exactly the same name is a bit confusing to me. I'd tend to call the utility function `redis_calculate_key` to avoid that.